File: py/IEC61970/Base/Wires/Cut.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106

from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.Length import Length
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Wires.ACLineSegment import ACLineSegment
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Wires.Switch import Switch


class Cut(Switch):
    def __init__(self) -> None:
        super().__init__()
        self.ac_line_segment = ACLineSegment()
        self.length_from_terminal1 = Length()

    def get_ac_line_segment(self) -> ACLineSegment:
        return self.ac_line_segment

    def get_length_from_terminal1(self) -> Length:
        return self.length_from_terminal1

    def set_ac_line_segment(self, new_val: ACLineSegment) -> None:
        self.ac_line_segment = new_val

    def set_length_from_terminal1(self, new_val: Length) -> None:
        self.length_from_terminal1 = new_val

File: java/IEC61970/Base/Wires/Cut.java
package IEC61970.Base.Wires;

import IEC61970.Base.Domain.Length;

/**
 * A cut separates a line segment into two parts. The cut appears as a switch
 * inserted between these two parts and connects them together. As the cut is
 * normally open there is no galvanic connection between the two line segment
 * parts. But it is possible to close the cut to get galvanic connection.
 * The cut terminals are oriented towards the line segment terminals with the same
 * sequence number. Hence the cut terminal with sequence number equal to 1 is
 * oriented to the line segment's terminal with sequence number equal to 1.
 * The cut terminals also act as connection points for jumpers and other equipment,
 * e.g. a mobile generator. To enable this, connectivity nodes are placed at the
 * cut terminals. Once the connectivity nodes are in place any conducting
 * equipment can be connected at them.
 * @author selaost1
 * @version 1.0
 * @created 25-Dec-2023 8:31:55 PM
 */
public class Cut extends Switch {

	/**
	 * The length to the place where the cut is located starting from side one of the
	 * cut line segment, i.e. the line segment Terminal with sequenceNumber equal to 1.
	 */
	public Length lengthFromTerminal1;
	/**
	 * The line segment to which the cut is applied.
	 */
	public ACLineSegment ACLineSegment;

	public Cut(){

	}
}//end Cut